docs/src/main/tut/auth.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: docs
+title: Authentication API
+---
+
+# Authentication API
+
+Github4s supports the [Authentication API](https://developer.github.com/v3/oauth_authorizations/). As a result,
+with github4s, you can:
+
+- [New authentication](#new-authentication)
+- [Authorize url](#authorize-url)
+- [Get access token](#get-access-token)
+
+The following examples assume the following imports:
+
+```tut:silent
+import github4s.Github
+import github4s.Github._
+import github4s.jvm.Implicits._
+import scalaj.http.HttpResponse
+// if you're using ScalaJS, replace occurrences of HttpResponse by SimpleHttpResponse
+//import github4s.js.Implicits._
+//import fr.hmil.roshttp.response.SimpleHttpResponse
+
+```
+
+They also make use of `cats.Id` but any type container implementing `MonadError[M, Throwable]` will
+do.
+
+Support for `cats.Id`, `cats.Eval` and `Future` (the only supported option for scala-js) are
+provided out of the box when importing `github4s.{js,jvm}.Implicits._`.
+
+## Authentication
+
+### New authentication
+
+
+ Call to request a new authorization given a basic authentication, the returned object Authorization includes an
+ access token
+
+- `username`: the username of the user
+- `password`: the password of the user
+- `scopes`: attached to the token
+- `note`: to remind you what the OAuth token is for
+- `client_id`: the 20 character OAuth app client key for which to create the token
+- `client_secret`: the 40 character OAuth app client secret for which to create the token
+- `headers`: optional user headers to include in the request
```

Comment:
I don't think this is necessary since headers are not specified when calling newAuth